### ISSUE TITLE
"Characters" on status bar displaying wrong number for some Unicode characters

### ISSUE THREAD

**Issue by wmjordan** [stefankueng/BowPad#144]
The number of characters displayed on the status bar, nowadays, is the number of bytes when select characters are encoded with UTF-8 encoding, not the number of human script characters, for instance, usual Chinese characters requires 3 UTF-8 bytes to encode, some Chinese characters require 4 bytes or more.

For performance considerations, it may not be a good idea to fix this in BowPad.
Shall we write a script to calculate the actual character numbers?

**Comment by LLZWWY** [comment]
Excuse me, where can I find this file tinyexpr.h?

**Comment by stefankueng** [comment]
or extend the "summary" dialog to include the character count?

**Comment by wmjordan** [comment]
Presenting this in the Summary dialog is not so appropriate, since it calculate statistics of the whole document, yet the counter component on the status bar counts selected characters.

**Comment by wmjordan** [comment]
`BowPad.SciGetSelText` has already converted UTF-8 encoded string into UTF-16 string. It is very easy to write a script to do count characters. In UTF-16, we just need to take care about characters that have code points above 0xFFFF (encoded with Surrogate Pairs, two `WORD`s) and the other 60 thousands of characters can be counted incrementally.

Here's the script and the demonstration. Notice that the last character in the selection has a Unicode of 0x20DDA, thus it needs two UTF-16 characters to encode it.

![image](https://user-images.githubusercontent.com/12182730/103716047-0d191580-4ffd-11eb-91e9-d03b282f5585.png)

If we select a single character, the script reports the Unicode of the character.

![image](https://user-images.githubusercontent.com/12182730/103719497-16f24700-5004-11eb-9e35-8ab6b49a939d.png)

[CountCharacters.zip](https://github.com/stefankueng/BowPad/files/5773479/CountCharacters.zip)

**Comment by stefankueng** [comment]
so you're ok with a plugin?

**Comment by stefankueng** [comment]
on another note: the icon is black, which makes it almost invisible in dark mode. Maybe another color would help.

**Comment by wmjordan** [comment]
I tested the plugin with English, German, Chinese and Japanese documents. No problem was found so far.

You are right about the color of the plugin. I have changed the color to dark red, so it looks better in the Dark Mode.

![image](https://user-images.githubusercontent.com/12182730/103835823-39e33080-50c2-11eb-8a58-d2c9796692a2.png)
[CountCharacters.zip](https://github.com/stefankueng/BowPad/files/5778841/CountCharacters.zip)
